vsphere/resource_vsphere_license_test.go

```diff
@@ -0,0 +1,225 @@
+package vsphere
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/license"
+)
+
+func TestAccVSphereLicenseBasic(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccVSpherePreLicenseBasicCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccVSphereLicenseDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseBasicConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseExists("vsphere_license.foo"),
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccVSphereLicenseInvalid(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseInvalidConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseNotExists("vsphere_license.foo"),
+				),
+				ExpectError: regexp.MustCompile("License is not valid for this product"),
+			},
+		},
+	})
+
+}
+
+func TestAccVSphereLicenseWithLabelsOnVCenter(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccVSpherePreLicenseBasicCheck(t)
+			testAccVspherePreLicenseESXiServerIsNotSetCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseWithLabelConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseWithLabelExists("vsphere_license.foo"),
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccVSphereLicenseWithLabelsOnESXiServer(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccVSpherePreLicenseBasicCheck(t)
+			testAccVspherePreLicenseESXiServerIsSetCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseWithLabelConfig(),
+				// This error will be thrown when ran against an ESXi server.
+				ExpectError: regexp.MustCompile("Labels are not allowed in ESXi"),
+			},
+		},
+	})
+
+}
+
+func testAccVspherePreLicenseESXiServerIsNotSetCheck(t *testing.T) {
+	key, err := strconv.ParseBool(os.Getenv("VSPHERE_TEST_ESXI"))
+	if err == nil && key {
+		t.Skip("VSPHERE_TEST_ESXI must not be set for this acceptance test")
+	}
+}
+func testAccVspherePreLicenseESXiServerIsSetCheck(t *testing.T) {
+	key, err := strconv.ParseBool(os.Getenv("VSPHERE_TEST_ESXI"))
+	if err != nil || !key {
+		t.Skip("VSPHERE_TEST_ESXI must be set to true for this acceptance test")
+	}
+}
+
+func testAccVSphereLicenseInvalidConfig() string {
+	// quite sure this key cannot be valid
+	return `resource "vsphere_license" "foo" {
```

Comment:
Hey @girishramnani, looks like this one is still indented - might be worthwhile to just move this to a constant too as nothing in it is being generated.